### ISSUE TITLE
[TF2] [Feature Request] Player should be awarded the kill if he pushed the enemy off to environmental hazards

### ISSUE THREAD

**Issue by agrastiOs** [ValveSoftware/Source-1-Games#2058]
Sometimes spies cloak on ledges over a pit, attempting to push enemies to their death.
But, they're not awarded the kill. They should be, as without their action the enemies wouldn't have die.

**Comment by hahagoni** [comment]
I don't think this would be a good change as it could possibly discourage people from doing their supposed job and being more useful

**Comment by agrastiOs** [comment]
That isn't a good logic. Same could be said about the Sandvich.

**Comment by hahagoni** [comment]
I don't quite understand your point. Explain

**Comment by agrastiOs** [comment]
What I mean, friendlies can use Sandviches to heal enemies and do nothing. 
While this can happen randomly. For example if a Scout jumps over the pit and accidentally smash into an enemy and fall down.
Including that, sometimes Spies can do pretty severe damage from the team. The route would be blocked and probably most of the enemy team would die. But that's not my point. The spies just were an example.

**Comment by Haze33E** [comment]
I think this should only apply to things like the flamethrower airblast, rocket/grenade explosion knockback, Loose Cannon knockback, the force of nature and the shortstop shove. It should show you killed them with the weapon you used to knock them to their death. Instead you just get the environmental death icon. Worse still environmental kills don't count towards a killstreak on killstreak weapons.

**Comment by agrastiOs** [comment]
That would be a problem with the player, though. Same could be said about new weapons or class reworks, or about a every new feature added. Everyone would be the same class/have the same weapon, and try to utilize the weapon's abilities.
